protocol/messenger_identity.go

```diff
@@ -106,8 +106,19 @@ func (m *Messenger) SaveSyncDisplayName(displayName string, clock uint64) error
 		return err
 	}
 
-	m.account.Name = displayName
-	return m.multiAccounts.SaveAccount(*m.account)
+	preferredNameClock, err := m.settings.GetSettingLastSynced(settings.PreferredName)
+	if err != nil {
+		return err
+	}
+	// When either the display name or preferred name changes, m.account.Name should be updated.
+	// However, a race condition can occur during BackupData, where m.account.Name could be incorrectly updated.
+	// The final value of m.account.Name depending on which backup message(BackedUpProfile/BackedUpSettings) arrives later.
+	// So we should check the clock of the preferred name and only update m.account.Name if it's older than the display name.
+	if preferredNameClock < clock {
```

Comment:
@qfrank this will never happen, cause `preferredNameClock` will be always newer than `clock`. 

The only place where we're calling this function from is `handleBackedUpProfile`. It is called only in case when user is recovering his Status account. In order to start recovering we usually call `StartNodeWithAccount` function which creates Status account and also sets details to `accounts` table with the clock set to the time of recovering which is always newer than the clock that refers to the backed-up profile data.

So, because of this condition, after recovering, when you restart the app we will see 3-words name instead of backed-up display name.

Plase either handle this other way or revert changes made in this function.

Reply:
@qfrank thinking now, one of the quickest things we can do to overcome this and keep the code you've added is to update `preferredNameClock` to `-1` if `ProcessBackedupMessages` in the node config is set to `true`, when `SaveAccountAndStartNodeWithKey` and `StartNodeWithAccountAndInitialConfig` are called.

Reply:
Discussed all these with @qfrank offline, so his point in setting the display name as the name in `accounts.sql` if the display name clock is newer than the preferred name clock (preferred name states for selected ens name) comes from our requirement "if a user has registered ens name, it has higher priority over display name".

But I am not sure if that should be the case for profiles recovered from backup.
What if the display name expires in the meantime?

Only for recovered Status profiles, I am more for an option to set the display name and if the user has an ens name registered for his profile he will need to go to the ens management part of the app and register it (in this case we don't need even to backup preferred name).

What do you think about it?

Reply:
> if i remember correctly we have a go routine which will check the state of ENS name. It's not friendly that user need to go to ens management part manually to register from user view

I agree with @qfrank 👍 
If there will be problems about this, we can open a separate issue and discuss it when ENS rework is done. It's planned in a few milestones afaik.